src/Symfony/Component/Messenger/Transport/Doctrine/Connection.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Doctrine;
+
+use Doctrine\DBAL\Connection as DriverConnection;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\Synchronizer\SingleDatabaseSynchronizer;
+use Doctrine\DBAL\Types\Type;
+
+/**
+ * @author Vincent Touzet <vincent.touzet@gmail.com>
+ */
+class Connection
+{
+    private $tableName;
+    private $redeliverTimeout;
+    private $debug;
+    private $driverConnection;
+
+    const STATUS_PENDING = 'pending';
+    const STATUS_DELIVERED = 'delivered';
+
+    public function __construct(string $tableName, DriverConnection $driverConnection, bool $debug = false, int $redeliverTimeout = 3600)
+    {
+        $this->tableName = $tableName;
+        $this->redeliverTimeout = $redeliverTimeout;
+        $this->driverConnection = $driverConnection;
+        $this->debug = $debug;
+    }
+
+    public function publish(string $body, array $headers): void
+    {
+        if ($this->debug) {
+            $this->setup();
```

Comment:
I agree with your earlier comment that we really need a system to "setup" the queues. Actually, I'm not sure they should be automatically set up in dev at all. #29476 talks about this as well. We should make that happen.

Reply:
Maybe I can work on the #29476 before this one ?

Reply:
We are almost there with #29476, if you could do the small changes, it would be great. The reasoning behind setting them up automatically in `dev` was the developer experience. Though, if we believe it's more of a source of troubles for later... happy to get rid of it.